src/Core/Billing/Services/Implementations/OrganizationSubscriptionService.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Bit.Core.AdminConsole.Entities;
+using Bit.Core.Enums;
+using Bit.Core.Models.Business;
+using Bit.Core.Services;
+using Bit.Core.Settings;
+using Microsoft.Extensions.Logging;
+using Stripe;
+using Customer = Stripe.Customer;
+using StaticStore = Bit.Core.Models.StaticStore;
+using Subscription = Stripe.Subscription;
+
+namespace Bit.Core.Billing.Services.Implementations;
+
+public class OrganizationSubscriptionService(
+    IGlobalSettings globalSettings,
+    ILogger<SubscriberService> logger,
+    IStripeAdapter stripeAdapter) : IOrganizationSubscriptionService
+{
+    private const string SecretsManagerStandaloneDiscountId = "sm-standalone";
+
+    public async Task<string> PurchaseOrganizationNoPaymentMethod(Organization org, StaticStore.Plan plan,
```

Comment:
❌ Just mentioning this here even though we discussed this on Slack, but this new service won't be necessary. The idea behind the `OrganizationBillingService` is that we should completely centralize all of the logic around purchasing organizations so that our team can manage the contract and logic. 

After [this](https://github.com/bitwarden/server/pull/4800) is merged, all of this can be accomplished with the following:
```csharp
var sale = OrganizationSale.From(organization, signup);
await organizationBillingService.Finalize(sale);
```
This will already handle all of the logic implemented here.